src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/createcase/CreateCaseMidEventHandler.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.createcase;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.NO;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+import static uk.gov.hmcts.reform.sscs.model.AppConstants.IBCA_BENEFIT_CODE;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
+import uk.gov.hmcts.reform.sscs.ccd.domain.BenefitType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicList;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DynamicListItem;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Entity;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.ccd.validation.address.PostcodeValidator;
+
+@Component
+@Slf4j
+public class CreateCaseMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private final PostcodeValidator postcodeValidator = new PostcodeValidator();
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && (callback.getEvent() == EventType.VALID_APPEAL_CREATED
+                || callback.getEvent() == EventType.NON_COMPLIANT
+                || callback.getEvent() == EventType.INCOMPLETE_APPLICATION_RECEIVED
+                || callback.getEvent() == EventType.CASE_UPDATED)
+                && Objects.nonNull(callback.getCaseDetails())
+                && Objects.nonNull(callback.getCaseDetails().getCaseData());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> errorResponse = new PreSubmitCallbackResponse<>(caseData);
+
+        if (isIbcaCase(caseData)) {
```

Comment:
if it's only for IBA cases, is it worth doing this check in the canHandle method?